### PR TITLE
add observer component

### PR DESCRIPTION
**Pull request by krancour**
This component watches worker and job pods in Kubernetes and reports their status via the Brigade API.

### CODE REVIEW

File: v2/internal/kubernetes/common.go [hunk 1]
```diff
@@ -10,3 +12,19 @@ const (
 	SecretTypeEvent          = "brigade.sh/event"
 	SecretTypeJobSecrets     = "brigade.sh/job"
 )
+
+func WorkerPodsSelector() string {
+	return labels.Set(
+		map[string]string{
+			LabelComponent: "worker",
```
Comment (radu-matei) [inline v2/internal/kubernetes/common.go:19]: #1174 addresses this, right?

File: v2/observer/workers_test.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second)
+	defer cancel()
+
+	go observer.syncWorkerPods(ctx)
+
+	// The informer needs a little time to get going. If we don't put a little
+	// delay here, we'll be adding, updating, and deleting pods before the
+	// informer gets cranking.
```
Comment (radu-matei) [inline v2/observer/workers_test.go:49]: This shouldn't be an issue in actual deployments of Brigade, right?

**Review by radu-matei — APPROVED** [review]

File: v2/internal/kubernetes/common.go (hunk 1 above)
Comment (krancour) [reply v2/internal/kubernetes/common.go:19]: Yes

File: v2/observer/workers_test.go (hunk 2 above)
Comment (krancour) [reply v2/observer/workers_test.go:49]: Correct

**Pull request merged**